StellarApp/Controllers/SolarViewController.swift

```diff
@@ -1,75 +1,189 @@
+    var portalChange: PortalChange = .reality {
+        didSet {
+            updateScene()
+        }
+    }
+
+    lazy var solarView: SolarView = {
+        return SolarView()
+    }()
+
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        // Set the view's delegate
         sceneView.delegate = self
         
-        // Show statistics such as fps and timing information
-        sceneView.showsStatistics = true
+        //sceneView.showsStatistics = true
+
+        updateScene()
         
-        // Create a new scene
-        let scene = SCNScene(named: "art.scnassets/ship.scn")!
+        //        Sound.playSound(sound: "background", format: "mp3")
         
-        // Set the scene to the view
-        sceneView.scene = scene
+
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(self.tapped(sender:)))
+        let pinchGestureRecognizer = UIPinchGestureRecognizer(target: self, action: #selector(self.pinch(pinch:)))
+        sceneView.addGestureRecognizer(tapGestureRecognizer)
+        sceneView.addGestureRecognizer(pinchGestureRecognizer)
+
+        setUpSolarView()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        var customTabFrame = self.tabBarController?.tabBar.frame
+        customTabFrame?.size.height = CGFloat(customTabBarHeight)
+        customTabFrame?.origin.y = self.view.frame.size.height - CGFloat(customTabBarHeight)
+        self.tabBarController?.tabBar.frame = customTabFrame!
+        tabBarController?.tabBar.barTintColor = .black
+        tabBarController?.tabBar.tintColor = .white
+    }
+
+    @objc func playPressed() {
+//        if playAnimation == .regular {
+//            playAnimation = .animation
+//        } else {
+//            playAnimation = .regular
+//        }
+//
+//
+//        switch playAnimation {
+//        case .regular:
+//
+//            sceneView.scene.rootNode.enumerateChildNodes { (node, stop ) in
+//                node.removeFromParentNode()
+//            }
+//
+//            sceneView.scene.rootNode.addChildNode(centerNode)
+//            Planet.getPlanets().forEach { centerNode.addChildNode($0) }
+//            sceneView.isUserInteractionEnabled = true
+//
+//        case .animation:
+//
+//            let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(tapped))
+//            let pinchGestureRecognizer = UIPinchGestureRecognizer(target: self, action: #selector(pinch))
+//
+//            sceneView.removeGestureRecognizer(tapGestureRecognizer)
+//            sceneView.removeGestureRecognizer(pinchGestureRecognizer)
+//            sceneView.scene.rootNode.enumerateChildNodes { (node, stop ) in
+//                node.removeFromParentNode()
+//            }
+//            sceneView.scene.rootNode.addChildNode(centerNode)
+//            MovedPlanet.getPlanets().forEach { centerNode.addChildNode($0) }
+//            sceneView.isUserInteractionEnabled = false
+//
+//        }
```

Comment:
commented out code needed?

Reply:
yes, its for the play button, before we click on it, all the planets are not moving, after we click it, all the planets will be moving around the sun

Reply:
but since last time you help me with code, we haven't talked about how to refactor these codes.